### PR TITLE
Allow short-circuiting `get_post_class` for performance

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
The PR refresh the final patch [37114.3.patch](https://core.trac.wordpress.org/attachment/ticket/37114/37114.3.patch) also update the docblock for `$classes` and `$class` similar to `post_class` filter.

Trac ticket: https://core.trac.wordpress.org/ticket/37114

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/post-template.php [hunk 1]
```diff
@@ -547,6 +547,21 @@ function get_post_class( $class = '', $post = null ) {
...
+	 * Default is all public taxonomies registered to the post type.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array    $taxonomies List of all public taxonomies to generate classes for.
+	 * @param string[] $classes    An array of post class names.
+	 * @param string[] $class      An array of additional class names added to the post.
+	 * @param int      $post_id    The post ID.
+	*/
+	$taxonomies = apply_filters( 'post_class_taxonomies', $taxonomies, $classes, $class, $post->ID );
```
Comment (peterwilsoncc) [inline src/wp-includes/post-template.php]:
```suggestion
	$taxonomies = apply_filters( 'wp_post_class_taxonomies', $taxonomies, $classes, $class, $post->ID );
```

namespace all the things: Core hasn't been great at this but it's nice to do for new items.

I'd also consider reordering the parameters:
* taxonomies
* post ID
* classes
* class

Comment (mukeshpanchal27) [reply src/wp-includes/post-template.php]: Thanks for the feedback. Change the order that make more sense.

**Comment by mukeshpanchal27** [comment]
@peterwilsoncc Can you please re-review the PR?

**Review by peterwilsoncc — APPROVED** [review]
Thank you, LGTM!

**Comment by peterwilsoncc** [comment]
Merged in https://core.trac.wordpress.org/changeset/54066 / 37b2a7fb14df0911836e63960e0ad263232aa82e

**Pull request closed without merging**